### DISCUSSION TITLE
Etherscan unable to verify Uniswap V2 clone contract

### DISCUSSION THREAD

**Discussion by LanceAddison** [Cyfrin/foundry-full-course-cu#2894] [Q&A] [answered]
I tried running this script in foundry to deploy my contract to Sepolia but when it got to the verification part it gave me an error saying the compiled byte-code does not match the existing byte-code. I made sure my Etherscan-API key was working so I'm really not sure what the issue could be. If someone could help me figure out this error I would appreciate it.

Command:
```
forge script script/DeployCrypticFlowRouter.s.sol:DeployCrypticFlowRouter --rpc-url $SEPOLIA_RPC_URL --account devKeyAccount1 --sender 0x3da99E808a621112b91d108De426439813cF6005 --broadcast --verify --etherscan-api-key $(ETHERSCAN_API_KEY) -vvvv
```

Error: 
```
==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.
Warning: We haven't found any matching bytecode for the following contracts: [0xe11f81accd8d5c1dcd91aade95eebea15f119195, 0xe50525881660c567fae6658d861aba8625ace1e1].

This may occur when resuming a verification, but the underlying source code or compiler version has changed.
Start verification for (1) contracts
Start verifying contract 0x4111561a99C99d3BcC5B588BA9444393C7adBaCe deployed on sepolia

Submitting verification for [src/libraries/CrypticFlowLibrary.sol:CrypticFlowLibrary] 0x4111561a99C99d3BcC5B588BA9444393C7adBaCe.
Submitted contract for verification:
        Response: OK
        GUID: ufidfqy7hk8sgjf67dwqzidft3mkjheydcxj8vsrsrbdmgzpf4
        URL: https://sepolia.etherscan.io/address/0x4111561a99c99d3bcc5b588ba9444393c7adbace
Contract verification status:
Response: NOTOK
Details: Pending in queue
Warning: Verification is still pending... (7 tries remaining)
Contract verification status:
Response: NOTOK
Details: Fail - Unable to verify. Compiled contract deployment bytecode does NOT match the transaction deployment bytecode.
Error: Failed to verify contract: Checking verification result failed: Contract failed to verify.

Transactions saved to: /Users/lanceaddison/Library/Mobile Documents/comappleCloudDocs/Code/Web3 Development/Personal Projects/CrypticFlow-DEX/broadcast/DeployCrypticFlowRouter.s.sol/11155111/run-latest.json

Sensitive values saved to: /Users/lanceaddison/Library/Mobile Documents/comappleCloudDocs/Code/Web3 Development/Personal Projects/CrypticFlow-DEX/cache/DeployCrypticFlowRouter.s.sol/11155111/run-latest.json

Error: Not all (0 / 1) contracts were verified!
make: *** [deployCrypticFlowRouter] Error 1
``` 

DeployCrypticFlowRouter contract:
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.25;

import {Script} from "forge-std/Script.sol";
import {CrypticFlowFactory} from "../src/CrypticFlowFactory.sol";
import {CrypticFlowRouter} from "../src/CrypticFlowRouter.sol";

contract DeployCrypticFlowRouter is Script {
    // change this address
    // address defaultTestContract = 0x7FA9385bE102ac3EAc297483Dd6233D62b3e1496;
    address feeToAddress = "myAddress";

    function run() external returns (CrypticFlowFactory, CrypticFlowRouter) {
        vm.startBroadcast();
        CrypticFlowFactory factory = new CrypticFlowFactory(feeToAddress);
        CrypticFlowRouter router = new CrypticFlowRouter(address(factory));
        vm.stopBroadcast();

        return (factory, router);
    }
}
```

**Comment by EngrPips** [comment]
This part used to be a tough one for me, too.

**Reply by LanceAddison** [reply, marked as answer]
Ok nevermind. I figured it out. 😅

I just added --via-ir to the end of the forge script command and it worked. I'm glad it worked but I have no idea why. 

I've only completed up to foundry advanced so I'm not sure if this is covered in later courses but it would be nice if there was a course all about this nuanced stuff like etherscan, foundry cast, slither, invariant tests(they're hard). Just somthing that covers most of the other stuff that isn't contract dev specifically. I'm sure I missed even more that could be covered but I constantly feel like I am being hindered by other steps that are part of the process like verifying on etherscan. I have had some sort of problem with contract verification every time I have tried to deploy a contract not just this one. Just a suggestion for @PatrickAlphaC / Cyfrin in general

**Reply by EngrPips** [reply]
Honestly, This is a good suggestion, but then no course can cover everything you need to know. You will learn some stuff through your personal struggles and suffering, and that is completely normal.

**Reply by LanceAddison** [reply]
Yeah very true. I'm sure to an extent stuff like etherscan and other tools that are helpful after or during writing a contract could be put in a course to get a more in depth understanding of how they work and how to use them. I definitely agree that suffer is part of the developer experience though. 🥲

**Reply by EngrPips** [reply]
Indeed.

**Comment by Nika250** [comment]
I'm glad it worked but I have no idea why.

**Reply by LanceAddison** [reply]
Did you also have the same issue and just added --via-ir and it fixed it?

**Reply by Nika250** [reply]
Yes

**Reply by EngrPips** [reply]
Awesome.

**Comment by Gektor22** [comment]
hello 🙌

**Reply by Gektor22** [reply]
> hello 🙌

ii

**Reply by EngrPips** [reply]
Howdy.
